components/collaboration/index.tsx

```diff
@@ -1,56 +1,123 @@
+import { IBrands } from "@/shared-libs/firestore/trendly-pro/models/brands";
 
-const getUniqueValues = (array: string[], key: number) => {
-  return ["All", ...new Set(array.map((item) => item[key]))];
-};
+interface ICollaborationAddCardProps extends ICollaboration {
+  name: string;
+  brandName: string;
+  paymentVerified?: boolean;
+  appliedCount?: number;
+  aiSuccessRate?: string;
+  id: string;
+  brandHireRate?: string;
+}
 
 const Collaboration = () => {
+  const getUniqueValues = (array: any[], key: string) => {
+    return ["All", ...new Set(array.map((item) => item[key]))];
+  };
   const [filterVisible, setFilterVisible] = useState(false);
   const [searchQuery, setSearchQuery] = useState("");
   const [selectedCategory, setSelectedCategory] = useState("");
   const [selectedJobType, setSelectedJobType] = useState("");
-  const [salaryRange, setSalaryRange] = useState([0, 999999]);
+  const [salaryRange, setSalaryRange] = useState([0, 10000]);
+  const [collabs, setCollabs] = useState<ICollaborationAddCardProps[]>([]);
   const { colors } = useTheme();
-  const styles = createStyles(colors);
-  console.log(colors.background);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      await signInAnonymously(AuthApp);
+      const fetchedBrands = await fetchBrands(); // Fetch brands and store in a variable
+      console.log("Fetched Brands: ", fetchedBrands);
+      await fetchCollabs(fetchedBrands); // Pass the fetched brands to fetchCollabs
+    };
+
+    fetchData();
+  }, []);
+
+  const fetchBrands = async () => {
+    const brandRef = collection(FirestoreDB, "brands");
+    const snapshot = await getDocs(brandRef);
+    const brandData = snapshot.docs.map((doc) => {
+      const id = doc.id;
+      const data = doc.data() as IBrands;
+      return { ...data, id };
+    });
+
+    const brandMap: {
+      [key: string]: { name: string; paymentMethodVerified: boolean };
+    } = {};
+    brandData.forEach((brand) => {
+      brandMap[brand.id] = {
+        name: brand.name,
+        paymentMethodVerified: brand.paymentMethodVerified || false,
+      };
+    });
+
+    return brandMap;
+  };
+
+  const fetchCollabs = async (brandsMap: {
+    [key: string]: { name: string; paymentMethodVerified: boolean };
+  }) => {
+    const collabRef = collection(FirestoreDB, "collaborations");
+    const snapshot = await getDocs(collabRef);
+    const data = snapshot.docs.map(
+      (doc) => ({ ...doc.data(), id: doc.id } as ICollaborationAddCardProps)
+    );
+
+    const collabsWithBrandNames = data.map((collab) => ({
+      ...collab,
+      brandName: brandsMap[collab.brandId].name || "Unknown Brand",
+      paymentVerified:
+        brandsMap[collab.brandId]?.paymentMethodVerified || false,
+    }));
+
+    setCollabs(collabsWithBrandNames);
+  };
 
   const toggleFilterModal = () => {
     setFilterVisible(!filterVisible);
   };
 
-  const filteredList = Collabs.filter((job) => {
+  console.log(
```

Comment:
is this console required? We can remove this console if not required to not leak data. Make sure console logs are removed everywhere except backend or firebase.